app/src/main/java/org/cru/godtools/ui/tools/ToolsAdapter.kt

```diff
@@ -53,11 +58,15 @@ class ToolsAdapter(lifecycleOwner: LifecycleOwner, private val dataModel: ToolsA
 
     override fun getItemViewType(position: Int) = when (getItem(position)?.type) {
         Tool.Type.LESSON -> VIEW_TYPE_LESSON
-        else -> VIEW_TYPE_TOOL
+        else -> when (isAllTools) {
+            true -> VIEW_TYPE_ALL_TOOL
+            false -> VIEW_TYPE_TOOL
+        }
```

Comment:
this when should be collapsed to be part of the parent when

Reply:
@frett not really sure how I would flatten this into one when, since I am checking two conditions?

Reply:
```
when {
   getItem(position)?.type == Tool.Type.LESSON -> VIEW_TYPE_LESSON
   isAllTools -> VIEW_TYPE_ALL_TOOLS
   ...
}
```